src/ring_theory/polynomial/cyclotomic/basic.lean

```diff
@@ -423,6 +423,47 @@ begin
            geom_sum_mul, prod_cyclotomic_eq_X_pow_sub_one h]
 end
 
+lemma cyclotomic_dvd_geom_sum_of_dvd (R) [comm_ring R] {d n : ℕ} (hdn : d ∣ n)
+  (hd : d ≠ 1) : cyclotomic d R ∣ geom_sum X n :=
+begin
+  suffices : (cyclotomic d ℤ).map (int.cast_ring_hom R) ∣ (geom_sum X n).map (int.cast_ring_hom R),
+  { have key := (map_ring_hom (int.cast_ring_hom R)).map_geom_sum X n,
+    simp only [coe_map_ring_hom, map_X] at key,
+    rwa [map_cyclotomic, key] at this },
+  apply map_dvd,
+  rcases n.eq_zero_or_pos with rfl | hn,
+  { simp },
+  rw ←prod_cyclotomic_eq_geom_sum hn,
+  apply finset.dvd_prod_of_mem,
+  simp [hd, hdn, hn.ne']
+end
+
+lemma X_pow_sub_one_mul_prod_cyclotomic_eq_X_pow_sub_one_of_dvd (R) [comm_ring R] {d n : ℕ}
+  (hd : d ∣ n) (hdn : d < n) :
```

Comment:
```suggestion
  (hd : d ∈ n.proper_divisors) :
```

Reply:
I tried this, but it was kind of inconvenient to use in the actual proofs

Reply:
How can be inconvenient? [nat.mem_proper_divisors](https://leanprover-community.github.io/mathlib_docs/number_theory/divisors.html#nat.mem_proper_divisors) says exactly the conditions are equivalent.

Reply:
oh! I naively assumed it'd be `≠ n`, instead of the nice `lt` condition which leads itself nicely to a proof. done